elastic/kibana
# [Infra UI] Smooth out spikey charts

## Summary

This PR smooths out spikey charts.

- Dotted line caused by period set in metricbeat to be greater than the date histogram interval determined by lens
<img width="1464" alt="image" src="https://github.com/elastic/kibana/assets/2767137/0f24b647-f000-4a09-a9a2-025efc56307a">


- Gap caused by a host that stopped shipping metrics
<img width="1460" alt="image" src="https://github.com/elastic/kibana/assets/2767137/6bb1c1bd-891a-42b4-bf3b-67d4abea7bb8">

- Before this change
<img width="1448" alt="image" src="https://github.com/elastic/kibana/assets/2767137/b4d2f0e2-698f-4339-bcea-e32451398a5b">

_The spikes are a result of the `period` set in metricbeat/integration being greater than the date histogram interval that Lens automatically sets according to the date range passed to the charts._

### How to test this PR

- Setup a local Kibana instance
- Configure `metricbeat.yml`, enabling the `system` module and setting the `period` with `1m`. Start a local metricbeat instance.
- Navigate to `Infrastructure` > `Hosts`
  - Verify if when date picker is set to < 1h hour interval the graphs will show the dotted lines
  - With 1h+ interval, there shouldn't be dotted lines, *unless* metricbeat is stopped and restarted after a few minutes
- Configure `metricbeat.yml`, enabling the `system` module and setting the `period` with `10s`. Restart the metricbeat instance. 
  - Verify if the charts maintain the existing behaviour
